scripts/west_commands/sdk.py

```diff
@@ -0,0 +1,525 @@
+
+
+            Installing SDK:
+
+                Run 'west sdk install' to install Zephyr SDK.
+
+                Set --version option to install the specified version of the SDK.
+                If not specified, the install version detects from "${ZEPHYR_BASE}/SDK_VERSION file.
+                SDKs older than 0.14.1 are not supported.
+
+                You can specify the installation directory with --install-dir.
+                If the specified version of the SDK is already installed,
+                the already installed SDK will be used regardless of the settings of
+                --install-dir.
+
+                Typically, Zephyr SDK archives contain only a directory named zephyr-sdk-<version>
+                at the top level.
+                Extract the archive to the home directory if not specified --install-dir.
+                In this case, SDK will install into ${HOME}/zephyr-sdk-<version>.
+                If specified --install-dir, The `zephyr-sdk-<version>` will rename and relocate
+                the specified location.
+
+                --interactive, --toolchains, --no-toolchains and --no-hosttools options
+                specify the behavior of the installer. Please see the description of each option.
+
+                --personal-access-token specifies the GitHub personal access token.
+                This helps to relax the limits on the number of REST API calls.
+
+                --api-url specifies the REST API endpoint for GitHub releases information
+                when installing the SDK from a different GitHub repository."""
+            ),
+        )
+
+        parser.add_argument(
+            "subcommand",
+            nargs="?",
+            default="list",
+            choices=["list", "install"],
+            help="select a subcommand. 'list' is the default.",
+        )
+        parser.add_argument(
+            "-v",
+            "--version",
+            default=None,
+            nargs="?",
+            metavar="SDK_VER",
+            help="specifies the install version of Zephyr SDK."
+            "If not specified, the install version detects from "
+            "${ZEPHYR_BASE}/SDK_VERSION file.",
+        )
+        parser.add_argument(
+            "-d",
+            "--install-dir",
+            default=None,
+            metavar="DIR",
+            help="specifies the install destination directory.",
+        )
+        parser.add_argument(
+            "-i",
+            "--interactive",
+            action="store_true",
+            help="launches installer in interactive mode. "
+            "--toolchains, --no-toolchains and --no-hosttools will be ignored if this option enabled.",
+        )
+        parser.add_argument(
+            "-t",
+            "--toolchains",
+            metavar="TC",
+            nargs="*",
+            help="specifies the installation of toolchains. "
+            "If neither is specified, all toolchains will be installed.",
```

Comment:
I assume this refers to cross toolchains for specific architectures, such as xtensa, arm, espressif, and so on.

However, I don't find that clear from this text.

Also the user of `neither` here is not clear if reading the help text when it's printed:
```
positional arguments:
  {list,install}        select a subcommand. 'list' is the default.

options:
  -h, --help            show this help message and exit
  -v [SDK_VER], --version [SDK_VER]
                        specifies the install version of Zephyr SDK.If not specified, the install version detects from ${ZEPHYR_BASE}/SDK_VERSION file.
  -d DIR, --install-dir DIR
                        specifies the install destination directory.
  -i, --interactive     launches installer in interactive mode. --toolchains, --no-toolchains and --no-hosttools will be ignored if this option enabled.
  -t [TC ...], --toolchains [TC ...]
                        specifies the installation of toolchains. If neither is specified, all toolchains will be installed.
  -T, --no-toolchains   do not install toolchains. --toolchains will be ignored if this option is enabled.
  -H, --no-hosttools    do not install host-tools.
  --personal-access-token PERSONAL_ACCESS_TOKEN
                        specify GitHub personal access token.
  --api-url API_URL     specifies the GitHub releases API endpoint for finding SDKs.

```

please be clearer on what exactly  neither refers to.

Reply:
I changed the `--toolchains` to a constraint that takes one or more arguments.
This change eliminates the zero-length arguments case and makes it clear that all toolchains are installed when the toolchains option is not specified, so I state so.

Reply:
> So please split list and install into two distinct sub-commands.

Fixed it.